cpp/src/cider/exec/processor/DefaultJoinHashTableBuilder.h

```diff
@@ -32,7 +32,12 @@ class DefaultJoinHashTableBuilder : public JoinHashTableBuilder {
  public:
   DefaultJoinHashTableBuilder(const ::substrait::JoinRel& joinRel,
                               const std::shared_ptr<JoinHashTableBuildContext>& context)
-      : joinRel_(joinRel), context_(context) {}
+      : joinRel_(joinRel), context_(context) {
+    // TODO(xinyi): pass some arguments that will decide hashtable type
+    // TODO(xinyi): 1. get the choosed hashtable type
+    // TODO(xinyi): 2. set the hashtable type
+    hashTable_ = std::unique_ptr<JoinHashTable>(new JoinHashTable());
```

Comment:
It seems that `hashTable_ = std::make_unique<JoinHashTable>();` is a better practice，I can change it in my next PR